transaction.go

```diff
@@ -84,51 +86,64 @@ func (e *EncryptedTransaction) print_decrypted() {
 	Dprintf("}\n")
 }
 
-func (en *Entry) verify(pks []ECPoint, CommCache ECPoint, RTokenCache ECPoint, eidx int, i int, debug string) bool {
+func (en *Entry) verify(pks []zksigma.ECPoint, CommCache zksigma.ECPoint, RTokenCache zksigma.ECPoint, eidx int, i int, debug string) bool {
 	// Check consistency proofs
-	if !VerifyConsistency(en.Comm, en.RToken, pks[i], en.CommConsistency) {
+	ok, err := en.CommConsistency.Verify(ZKLedgerCurve, en.Comm, en.RToken, pks[i])
+	if !ok {
 		Dprintf(" [%v] ETX %v Failed verify consistency comm entry %v %#v\n", debug, eidx, i, en)
+		Dprintf("  [%v] %s", debug, err.Error())
 		return false
 	}
-	if !VerifyConsistency(en.CommAux, en.BAux, pks[i], en.AuxConsistency) {
+	ok, err = en.AuxConsistency.Verify(ZKLedgerCurve, en.CommAux, en.BAux, pks[i])
+	if !ok {
 		Dprintf(" [%v] ETX %v Failed verify consistency aux entry %v\n", debug, eidx, i)
+		Dprintf("  [%v] %s", debug, err.Error())
 		return false
 	}
 	// Check Proof of Assets
-	Base1 := en.CommAux.Add(CommCache.Add(en.Comm).Neg())
-	Result1 := en.BAux.Add(RTokenCache.Add(en.RToken).Neg())
-	Result2 := en.CommAux.Add(en.Comm.Neg())
-	if !VerifyDisjunctive(Base1, Result1, EC.H, Result2, en.Assets) {
+	Base1 := ZKLedgerCurve.Add(en.CommAux, ZKLedgerCurve.Neg(ZKLedgerCurve.Add(CommCache, en.Comm)))
+	Result1 := ZKLedgerCurve.Add(en.BAux, ZKLedgerCurve.Neg(ZKLedgerCurve.Add(RTokenCache, en.RToken)))
+	Result2 := ZKLedgerCurve.Add(en.CommAux, ZKLedgerCurve.Neg(en.Comm))
+	ok, err = en.Assets.Verify(ZKLedgerCurve, Base1, Result1, ZKLedgerCurve.H, Result2)
+	if !ok {
 		fmt.Printf("  [%v] %v/%v Base1: %v\n", debug, eidx, i, Base1)
 		fmt.Printf("  [%v] %v/%v Result1: %v\n", debug, eidx, i, Result1)
 		fmt.Printf("  [%v] %v/%v Result2: %v\n", debug, eidx, i, Result2)
 		fmt.Printf("  [%v] ETX %v Failed verify left side of proof of assets entry %v\n", debug, eidx, i)
+		fmt.Printf("  [%v] %s", debug, err.Error())
 		return false
 	}
 	//   Range Proof
-	if !RangeProverVerify(en.CommAux, en.RP) {
+	ok, err = en.RP.Verify(ZKLedgerCurve, en.CommAux)
+	if !ok {
 		Dprintf("  [%v] %v/%v Range Proof: %v\n", debug, eidx, i, en.RP)
 		Dprintf("  [%v] ETX %v Failed verify the range proof on CommAux %v\n", debug, eidx, i)
+		Dprintf("  [%v] %s", debug, err.Error())
 		return false
 	}
 	return true
 }
 
-func (e *EncryptedTransaction) Verify(pks []ECPoint, CommCache []ECPoint, RTokenCache []ECPoint, debug string) bool {
+func (e *EncryptedTransaction) Verify(pks []zksigma.ECPoint, CommCache []zksigma.ECPoint, RTokenCache []zksigma.ECPoint, debug string) bool {
 	if e.skipVerify {
 		return true
 	}
-	g := GSPFS{curve: EC.C, ExponentPrime: EC.N, Generator: EC.H}
 	// Issuance
 	if e.Type == Issuance {
 		en := &e.Entries[e.Sender]
+		e.print_decrypted()
```

Comment:
is this supposed to be here?

Reply:
I think this is an artifact of Willy's code. Probably just to see some terminal output of what is happening. Does not need to be passed around anywhere so I don't think it is a big issue.